handlers/secrets.go

```diff
@@ -71,33 +71,56 @@ func UpdateSecrets(request requests.CreateFunctionRequest, deployment *v1beta1.D
 		}
 	}
 
-	if len(secretVolumeProjections) > 0 {
-		volumeName := fmt.Sprintf("%s-projected-secrets", request.Service)
-		projectedSecrets := apiv1.Volume{
-			Name: volumeName,
-			VolumeSource: apiv1.VolumeSource{
-				Projected: &apiv1.ProjectedVolumeSource{
-					Sources: secretVolumeProjections,
-				},
+	volumeName := fmt.Sprintf("%s-projected-secrets", request.Service)
+	projectedSecrets := apiv1.Volume{
+		Name: volumeName,
+		VolumeSource: apiv1.VolumeSource{
+			Projected: &apiv1.ProjectedVolumeSource{
+				Sources: secretVolumeProjections,
 			},
+		},
+	}
+
+	// remove the existing secrets volume, if we can find it. The update volume will be
+	// added below
+	existingVolumes := deployment.Spec.Template.Spec.Volumes
+	for i, v := range existingVolumes {
+		if v.Name == volumeName {
+			existingVolumes = append(existingVolumes[:i], existingVolumes[i+1:]...)
```

Comment:
I think this code should be rewritten to express what it is doing. It's removing an element.

`removeVolume()` would do.